src/Bicep.Core/Semantics/SemanticModel.cs

```diff
@@ -188,10 +189,33 @@ public IEnumerable<IDiagnostic> GetAllDiagnostics()
 
         private IEnumerable<IDiagnostic> AssembleDiagnostics()
         {
-            return GetParseDiagnostics()
-            .Concat(GetSemanticDiagnostics())
-            .Concat(GetAnalyzerDiagnostics())
-            .OrderBy(diag => diag.Span.Position);
+            var diagnostics = GetParseDiagnostics()
+                .Concat(GetSemanticDiagnostics())
+                .Concat(GetAnalyzerDiagnostics())
+                .OrderBy(diag => diag.Span.Position);
+            List<IDiagnostic> updatedDiagnostics = new List<IDiagnostic>();
+
+            var disableNextLineDiagnosticsDirectivesCache = SourceFile.DisableNextLineDiagnosticDirectivesCache;
+            foreach (IDiagnostic diagnostic in diagnostics)
+            {
+                (int diagnosticLine, _) = TextCoordinateConverter.GetPosition(SourceFile.LineStarts, diagnostic.Span.Position);
+
+                if (diagnosticLine == 0)
+                {
+                    updatedDiagnostics.Add(diagnostic);
+                    continue;
+                }
+
+                if (disableNextLineDiagnosticsDirectivesCache.TryGetValue(diagnosticLine - 1, out var value) &&
+                    value.diagnosticCodes.Contains(diagnostic.Code))
+                {
+                    continue;
+                }
```

Comment:
How do we decide which diagnostic messages can/can't be filtered out? For example - we shouldn't allow any core compiler errors to be filtered out, but (I think) want to allow core compiler warnings to be filtered. For linter issues, we might want to allow filtering of both warnings and errors?

Reply:
The current implementation doesn't allow any errors to be filtered out. 

Should it be changed to allow filtering of all but core compiler errors? i.e. the ones which have DiagnosticLevel set to error and start with BCP?

Reply:
Yes, but we may need a more robust mechanism than just checking the code prefix. (Unless we can guarantee that no linter rule will ever produce a BCP* diagnostic code.)

Reply:
I wouldn't want to rely on the naming convention alone. Would checking `diagnostic is AnalyzerDiagnostic` be sufficient here to differentiate?

If it's going to be tricky, another option is to just block overriding **all** errors, and revisit this in another PR.

Reply:
Btw, it looks like it's possible to suppress a compiler error right now. That we should block before the PR merged because it can break the compiler (should have negative test cases to ensure we never regress) and could result in us generating an invalid template.

Reply:
Ah I totally missed adding check in SemanticModel to block core compiler error suppression. Thanks for pointing it out.

I updated both SemanticModel and BicepCodeActionHandler to block core compiler error suppression. All other diagnostics can be suppressed via #disable-next-line directive. Also added bunch of tests.

https://github.com/Azure/bicep/commit/0fd16670d93a225c93b56a79a3e80362f299a483

Let me know if it looks good or if I am missing something,